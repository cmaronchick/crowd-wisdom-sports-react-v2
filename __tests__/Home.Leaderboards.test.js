import React from 'react';
import { BrowserRouter as Router, Switch, Route, Link } from "react-router-dom";
import HomeLeaderboards from '../src/components/Home.Leaderboards'
import renderer from 'react-test-renderer'

const testLeaderboardData = {
    _id: "5df31318b814f591b891cf67",
    gameWeek:15,
    season:"reg",
    sport:"nfl",
    year:2019,
    weekly:{
        crowd:{
            winner:{
                correct:11,
                push:0,
                percentage:0.6875
            },
            spread:{
                correct:6,
                push:1,
                percentage:0.4
            },
            total:{
                correct:9,
                push:0,
                percentage:0.5625
            },
            predictionScore:52,
            totalGames:16
        },users:[
            {
                username:"cmaronchick",
                winner:{correct:13,push:0,bullseyes:1},
                spread:{correct:11,push:1,bullseyes:3},
                total:{correct:6,push:0,bullseyes:0},
                stars:{wagered:30,net:9,roi:0.3},
                predictionScore:64,totalPredictions:16,
                preferred_username:"cmaronchick"
            },
            {
                username:"cmaronchick-hotmail",winner:{correct:10,push:0,bullseyes:1},spread:{correct:9,push:1,bullseyes:2},total:{correct:9,push:0,bullseyes:0},stars:{wagered:0,net:0,roi:null},predictionScore:59,totalPredictions:16,preferred_username:"CBSPetePrisco"
            },
            {
                username:"Facebook_10205896813734452",winner:{correct:53,push:0,bullseyes:8},spread:{correct:35,push:1,bullseyes:2},total:{correct:37,push:1,bullseyes:2},stars:{wagered:0,net:0,roi:null},predictionScore:262,totalPredictions:78,preferred_username:"ElliePeterson"
            }
        ]
    }
}

let fetchingLeaderboards = false,
    selectedLeaderboard = "overall",
    sport = "nfl",
    year = 2019,
    week = 15,
    season = "reg",
    handleSwitchLeaderboard = () => {},
    handleOnUserClick = () => {}

describe('it should present the Home Leaderboards', () => {
    
    test('present a blank leaderboard when there is no data', () => {
        const homeleaderboardsrender = renderer.create(<HomeLeaderboards />).toJSON();
        
        console.log({homeleaderboardsrender});
        expect(homeleaderboardsrender).toMatchSnapshot();
    })
    test('when sport and data are missing', () => {
        const homeleaderboardsrender = renderer.create(<Router><HomeLeaderboards year={year} week={week} season={season} /></Router>).toJSON();
        console.log({homeleaderboardsrender});
        expect(homeleaderboardsrender).toMatchSnapshot();
    })
    test('when everything but data is provided', () => {
        const homeleaderboardsrender = renderer.create(<Router><HomeLeaderboards sport={sport} year={year} week={week} season={season} /></Router>).toJSON();
        console.log({homeleaderboardsrender});
        expect(homeleaderboardsrender).toMatchSnapshot();
    })
    test('present a leaderboard when there is valid data', () => {
        const homeleaderboardsrender = renderer.create(<Router><HomeLeaderboards sport={sport} year={year} week={week} season={season} overallLeaderboardData={testLeaderboardData} /></Router>).toJSON();
        console.log({homeleaderboardsrender});
        expect(homeleaderboardsrender).toMatchSnapshot();
    })

})
