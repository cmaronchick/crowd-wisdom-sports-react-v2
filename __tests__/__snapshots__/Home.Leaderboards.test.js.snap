// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it should present the Home Leaderboards present a blank leaderboard when there is no data 1`] = `
<div>
  Loading Leaderboard
</div>
`;

exports[`it should present the Home Leaderboards present a leaderboard when there is valid data 1`] = `
<div
  className="homeLeaderboards"
>
  <button
    className="btn btn-primary"
    disabled={false}
    onClick={[Function]}
    type="button"
  >
    Show 
    Overall
     Leaderboard
  </button>
  <table
    className="rwd-table"
  >
    <thead>
      <tr>
        <th
          className="rank"
        >
          <span
            className="full abbrev"
          >
            Rank
          </span>
        </th>
        <th
          className="entryowner"
        >
          <span
            className="full"
          >
            Username
          </span>
        </th>
        <th
          className="Record"
        >
          <span
            className="full"
          >
            Record
          </span>
        </th>
        <th
          className="total"
        >
          <span
            className="full"
          >
            Score
          </span>
        </th>
      </tr>
    </thead>
    <tbody>
      <tr
        className=" alt-tr"
      >
        <td
          data-th="Rank"
        >
          1
        </td>
        <td
          data-th="Username"
        >
          <a
            href="/nfl/games/2019/reg/15?compareUsername=cmaronchick"
            onClick={[Function]}
          >
            cmaronchick
          </a>
        </td>
        <td
          data-th="Record"
        >
          30
          -
          17
        </td>
        <td
          data-th="Score"
        >
          64
          <div
            className="leaderboard userScoreDetails"
          >
            S/U: 
            13
            <br />
            ATS: 
            11
            <br />
            O/U: 
            6
          </div>
        </td>
      </tr>
      <tr
        className={null}
      >
        <td
          data-th="Rank"
        >
          2
        </td>
        <td
          data-th="Username"
        >
          <a
            href="/nfl/games/2019/reg/15?compareUsername=CBSPetePrisco"
            onClick={[Function]}
          >
            CBSPetePrisco
          </a>
        </td>
        <td
          data-th="Record"
        >
          28
          -
          19
        </td>
        <td
          data-th="Score"
        >
          59
          <div
            className="leaderboard userScoreDetails"
          >
            S/U: 
            10
            <br />
            ATS: 
            9
            <br />
            O/U: 
            9
          </div>
        </td>
      </tr>
      <tr
        className=" alt-tr"
      >
        <td
          data-th="Rank"
        >
          3
        </td>
        <td
          data-th="Username"
        >
          <a
            href="/nfl/games/2019/reg/15?compareUsername=ElliePeterson"
            onClick={[Function]}
          >
            ElliePeterson
          </a>
        </td>
        <td
          data-th="Record"
        >
          125
          -
          107
        </td>
        <td
          data-th="Score"
        >
          262
          <div
            className="leaderboard userScoreDetails"
          >
            S/U: 
            53
            <br />
            ATS: 
            35
            <br />
            O/U: 
            37
          </div>
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;

exports[`it should present the Home Leaderboards when everything but data is provided 1`] = `
<div>
  Loading Leaderboard
</div>
`;

exports[`it should present the Home Leaderboards when sport and data are missing 1`] = `
<div>
  Loading Leaderboard
</div>
`;
